### PR TITLE
EVG-7760: add option to merge user data parts

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-7760

Build needs to use the distro user data to run a powershell script. In the cloud-init docs, the proper way to use multiple separate shell scripts in user data is to [submit the user data in multipart format](https://cloudinit.readthedocs.io/en/20.1/topics/format.html#mime-multi-part-archive). This works fine on Linux machines. However, if you're on Windows and happen to use multiple of the same shell script type (e.g. two powershell scripts), it will fail to run any of them (so the host will never provision). Funnily enough, if you specify a cmd.exe script and a powershell script, this is fine - it's only if you specify multiple of the same script type. There's no documentation on this behavior for Windows , and the bug is reproducible if you specify two separate powershell scripts in Terraform.

The solution I made was add an option to just append the distro user data to the existing host provisioning user data script so we use one combined powershell script. If the distro user data script has a problem, appending the script still shouldn't affect host provisioning, because the commands in the distro user data will run after all the evergreen host provisioning script runs.

## Changes
* Add option to merge user data parts into a single part.
* Drive-by fix for some JavaScript errors on distro page.
* Add to local evergreen data so you can save SSH keys.

### CODE REVIEW

File: cloud/user_data.go [hunk 1]
```diff
@@ -130,33 +135,59 @@ func parseUserDataContentType(userData string) (string, error) {
...
 	if h.Distro.BootstrapSettings.Method != distro.BootstrapMethodUserData {
-		return customScript, nil
+		return custom, nil
 	}
 	settings := env.Settings()
 
 	creds, err := h.GenerateJasperCredentials(ctx, env)
 	if err != nil {
-		return customScript, errors.Wrapf(err, "problem generating Jasper credentials for host '%s'", h.Id)
+		return custom, errors.Wrapf(err, "problem generating Jasper credentials for host '%s'", h.Id)
```
Comment (tychoish) [inline cloud/user_data.go:155]: why return the script and an error?

**Comment by john-m-liu** [comment]
You've probably considered this already, but what's stopping us from writing the script so that it's a single one that does what all the parts need to do? I'm having some trouble understanding the problem

**Comment by tychoish** [comment]
provisioning and boot strapping hosts are done in user data scripts supplied by the system not the configuration so if someone *also* wants to provision something via user data, then we have to merge their script with our script.

**Comment by john-m-liu** [comment]
Does it work to provisioning in userdata? By provision I assume you mean run the distro's setup script

**Comment by tychoish** [comment]
also evergreen's agent/monitor bootstrapping.

File: cloud/user_data.go (hunk 1 above)
Comment (Kimchelly) [reply cloud/user_data.go]: It doesn't really matter what you return here, but I changed it to return "" anyways.

**Pull request merged**